core/src/test/scala/unit/kafka/log/LogCleanerParameterizedIntegrationTest.scala

```diff
@@ -121,9 +121,9 @@ class LogCleanerParameterizedIntegrationTest extends AbstractLogCleanerIntegrati
     // Set the last modified time to an old value to force deletion of old segments
     val endOffset = log.logEndOffset
     log.logSegments.foreach(_.lastModified = time.milliseconds - (2 * retentionMs))
-    TestUtils.waitUntilTrue(() => log.logStartOffset == endOffset,
+    TestUtils.waitUntilTrue(() => 1 == log.numberOfSegments,
       "Timed out waiting for deletion of old segments")
-    assertEquals(1, log.numberOfSegments)
+    assertEquals(log.logStartOffset, endOffset)
```

Comment:
I'm re-triggering the CI since there are 2 jobs not finished. Meanwhile, could you help explain why we need these 2 changes in v3.6, but not in v3.7/trunk?

Reply:
The tests is flaky post landing the PR in trunk: 

https://ge.apache.org/scans/tests?search.relativeStartTime=P28D&search.rootProjectNames=kafka&search.timeZoneId=Europe%2FBerlin&tests.container=kafka.log.LogCleanerParameterizedIntegrationTest

We have to raise a similar fix in trunk.

Reply:
Raised #15787 PR to fix it in trunk